lib/internal/modules/esm/resolve.js

```diff
@@ -940,7 +940,8 @@ function resolveAsCommonJS(specifier, parentURL) {
     } else if (isBareSpecifier(specifier)) {
       // If it is a bare specifier return the relative path within the
       // module
-      const pkg = StringPrototypeSlice(specifier, 0, StringPrototypeIndexOf(specifier, '/'));
+      const i = StringPrototypeIndexOf(specifier, '/');
+      const pkg = i === -1 ? specifier : StringPrototypeSlice(specifier, 0, i);
```

Comment:
Should this handle scopes?

```suggestion
      let i = StringPrototypeSlice(specifier, 0, StringPrototypeIndexOf(specifier, '/'));
      if (i !== -1 && specifier[0] === '@')
        i = StringPrototypeIndexOf(specifier, '/', i + 1);
      const pkg = i === -1 ? specifier : StringPrototypeSlice(specifier, 0, i);
```

Reply:
I don't think so, why would that matters? I'll add a test though